MobileSyncExplorerKotlinTemplate/app/src/main/java/com/salesforce/mobilesyncexplorerkotlintemplate/model/contacts/ContactObject.kt

```diff
@@ -0,0 +1,134 @@
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.mobilesyncexplorerkotlintemplate.model.contacts
+
+import com.salesforce.androidsdk.mobilesync.util.Constants
+import com.salesforce.mobilesyncexplorerkotlintemplate.core.extensions.optStringOrNull
+import com.salesforce.mobilesyncexplorerkotlintemplate.core.salesforceobject.*
+import org.json.JSONObject
+
+/**
+ * An abstraction and runtime data model of a Contact Salesforce Standard Object.
+ *
+ * Note how this is not a [SalesforceObject]. [SalesforceObject]s are _mutable_ which goes against
+ * Jetpack Compose guidelines to make state objects immutable.
+ */
+data class ContactObject
+@Throws(ContactValidationException::class) constructor(
+    val firstName: String?,
+    val lastName: String,
+    val title: String?,
+    val department: String?,
+) : SObject {
+    init {
+        validateLastName(lastName)
+        validateFirstName(firstName)
+    }
+
+    override val objectType: String = Constants.CONTACT
+
+    val fullName = formatFullName(firstName = firstName, lastName = lastName)
+
+    override fun JSONObject.applyObjProperties() = this.apply {
+        putOpt(KEY_FIRST_NAME, firstName)
+        putOpt(KEY_LAST_NAME, lastName)
+        putOpt(KEY_TITLE, title)
+        putOpt(KEY_DEPARTMENT, department)
+        putOpt(Constants.NAME, fullName)
+    }
+
+    companion object : SObjectDeserializerBase<ContactObject>(objectType = Constants.CONTACT) {
+        const val KEY_FIRST_NAME = "FirstName"
+        const val KEY_LAST_NAME = "LastName"
+        const val KEY_TITLE = "Title"
+        const val KEY_DEPARTMENT = "Department"
+
+        @Throws(CoerceException::class)
+        override fun buildModel(fromJson: JSONObject): ContactObject = try {
+            ContactObject(
+                firstName = fromJson.optStringOrNull(KEY_FIRST_NAME),
+                lastName = fromJson.optString(KEY_LAST_NAME),
+                title = fromJson.optStringOrNull(KEY_TITLE),
+                department = fromJson.optStringOrNull(KEY_DEPARTMENT),
+            )
+        } catch (ex: ContactValidationException) {
+            when (ex) {
+                ContactValidationException.LastNameCannotBeBlank -> InvalidPropertyValue(
```

Comment:
missing `is`?

Reply:
Since it's specific to one case, `LastNameCannotBeBlank` is an `object` and doesn't need the `is` 👍